src/syntax.ml

```diff
@@ -209,9 +324,11 @@ and delimited_inline_element_list :
     parent_markup:[< Token.t ] ->
     parent_markup_location:Loc.span ->
     requires_leading_whitespace:bool ->
+    context:inline_context ->
```

Comment:
This bit of threaded state is annoying. It's also very similar to the `requires_leading_whitespace` state - I wonder if the two could be combined into a single `Context.t` or similar? I suppose the new bit is relevant to both `delimited_inline_element_list` _and_ `inline_element` so maybe not worth it? It's also possibly worth considering getting rid of the context and just checking in the `light_table` code after we've called this function that the start and end lines are the same?

Reply:
I had a quick go: https://github.com/jonludlam/odoc-parser/commit/5c49f986e1094617d566a7a1a20b0106ff954ff3 - I think it's a bit less invasive and keeps the check within the `light_table_row` function. What do you think?

Reply:
It's also my opinion that having the check in `light_table_row` makes the code clearer.